### PR TITLE
Fix thumb ADR instruction alignment

### PR DESCRIPTION
**Pull request by mailwl**
Only one instruction fixed, not sure if another pc-based instructions must be aligned, too.

PR fixes the game **Conception 2: Children of the Seven Stars**

### CODE REVIEW

File: src/core/arm/dyncom/arm_dyncom_interpreter.cpp [hunk 1]
```diff
@@ -3956,8 +3956,11 @@ unsigned InterpreterMainLoop(ARMul_State* cpu) {
             add_inst* const inst_cream = (add_inst*)inst_base->component;
 
             u32 rn_val = RN;
-            if (inst_cream->Rn == 15)
+            if (inst_cream->Rn == 15) {
                 rn_val += 2 * cpu->GetInstructionSize();
+                if(cpu->TFlag)
```
Comment (lioncash) [inline src/core/arm/dyncom/arm_dyncom_interpreter.cpp]: space after the if

Comment (lioncash) [inline src/core/arm/dyncom/arm_dyncom_interpreter.cpp]: This if statement is unnecessary. You can just leave the AND in place.

Comment (lioncash) [inline src/core/arm/dyncom/arm_dyncom_interpreter.cpp]: Something is wrong if code is reaching here and the T flag is set. Thumb mode is not capable of encoding PC as register.

Comment (lioncash) [inline src/core/arm/dyncom/arm_dyncom_interpreter.cpp]: The branch here isn't necessary, you can just leave the AND

**Comment by JayFoxRox** [comment]
(ARM DDI 0406) says (about ADR):
```
if ConditionPassed() then
  EncodingSpecificOperations();
  result = if add then (Align(PC,4) + imm32) else (Align(PC,4) - imm32);
  if d == 15 then // Can only occur for ARM encodings
    ALUWritePC(result);
  else
    R[d] = result;
```

So the align seems to happen before the addition and it seems to happen regardless of ARM / Thumb mode.
Most (all?) other places probably use [CHECK_READ_REG15_WA](https://github.com/citra-emu/citra/blob/master/src/core/arm/dyncom/arm_dyncom_run.h#L33) already which does the align.

**Comment by mailwl** [comment]
changed to macro

**Comment by lioncash** [comment]
@mailwl This should also be done with SUB as well.

@JayFoxRox ADR is a pseudoinstruction which the assembler translates to an ADD or a SUB depending on the offset.

**Comment by lioncash** [comment]
Commit message should contain `alignment` instead of `align`. LGTM after that.

**Comment by wwylele** [comment]
I don't know about ARM instructions, but I saw some other instructions such as ADC, CMN have similar pattern - should them also be fixed?

**Comment by lioncash** [comment]
No, the other data-processing instructions don't require a fixed alignment. ADD and SUB are a special case, because ADR generates a PC-relative offset which needs to be word-aligned when the assembler decides which one the two instructions to generate.

**Comment by MerryMage** [comment]
The alignment should only happen on ADD/SUB R15 with an immediate.
ADD/SUB with a register operand do not perform the align.
(And with a regsiter-shifted register it's unpredictable.)

**Pull request merged**